### PR TITLE
enhance: remove non-main content from scraped HTML

### PR DESCRIPTION
**Pull request by iwilltry42**
.. so we later send less data to the LLM for advanced cleaning.
This is basically a 1:1 translation of what firecrawl does: https://github.com/mendableai/firecrawl/blob/bd928b1512ff08ba435f8ec23581f44c489ce84a/apps/api/src/scraper/scrapeURL/lib/removeUnwantedElements.ts#L6

Note: the dep update in knowledge is just something that caught my eye when I had the IDE open

### CODE REVIEW

**Review by thedadams — APPROVED** [review]

File: knowledge/data-sources/website/clean.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	"header", "footer", "nav", "aside", ".header", ".top", ".navbar", "#header",
+	".footer", ".bottom", "#footer", ".sidebar", ".side", ".aside", "#sidebar",
+	".modal", ".popup", "#modal", ".overlay", ".ad", ".ads", ".advert", "#ad",
+	".lang-selector", ".language", "#language-selector", ".social", ".social-media",
+	".social-links", "#social", ".menu", ".navigation", "#nav", ".breadcrumbs",
+	"#breadcrumbs", "#search-form", ".search", "#search", ".share", "#share",
+	".widget", "#widget", ".cookie", "#cookie",
+}
+
+func RemoveNonMainContent(doc *goquery.Selection) (*goquery.Selection, error) {
```
Comment (StrongMonkey) [inline knowledge/data-sources/website/clean.go:18]: this actually nerver return an error...

File: knowledge/data-sources/website/colly.go [hunk 2]
```diff
@@ -56,7 +56,13 @@ func scrape(ctx context.Context, converter *md.Converter, logOut *logrus.Logger,
 
 		logOut.Infof("scraping %s", e.Request.URL.String())
 		visited[e.Request.URL.String()] = struct{}{}
-		markdown := converter.Convert(e.DOM)
+
+		cleaned, err := RemoveNonMainContent(e.DOM)
+		if err != nil {
```
Comment (StrongMonkey) [inline knowledge/data-sources/website/colly.go:61]: same, no need to return err

**Review by StrongMonkey — APPROVED** [review]
Just one nit

File: knowledge/data-sources/website/clean.go (hunk 1 above)
Comment (iwilltry42) [reply knowledge/data-sources/website/clean.go]: True, I removed that part :D

**Pull request merged**